### PR TITLE
Replace a few stackallocs with collection expressions

### PR DESCRIPTION
**Pull request by stephentoub**

### CODE REVIEW

**Review by danmoseley — APPROVED** [review]

**Comment by stephentoub** [comment]
Thanks, @danmoseley, but CI is down and didn't run on this pr

**Comment by danmoseley** [comment]
Oh nuts... was there a way I could have determined that? I thought it was all green - but I'm on a phone.

**Comment by danmoseley** [comment]
Oh, expanding it I see the legs are missing.

**Comment by tannergooding** [comment]
I think this getting merged introduce a build break:
> src\libraries\System.Security.Cryptography\src\System\Security\Cryptography\Pbkdf2Implementation.Windows.cs(127,25): error CS0165: (NETCORE_ENGINEERING_TELEMETRY=Build) Use of unassigned local variable 'symmetricKeyMaterialLength'

File: src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/Pbkdf2Implementation.Windows.cs [hunk 1]
```diff
@@ -60,8 +60,7 @@ internal static partial class Pbkdf2Implementation
             if (password.IsEmpty)
             {
                 // CNG won't accept a null pointer for the password.
-                symmetricKeyMaterial = stackalloc byte[1];
-                symmetricKeyMaterialLength = 0;
```
Comment (kunalspathak) [inline src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/Pbkdf2Implementation.Windows.cs:64]: @stephentoub - this change is causing build errors:

```
D:\a\_work\1\s\src\libraries\System.Security.Cryptography\src\System\Security\Cryptography\Pbkdf2Implementation.Windows.cs(127,25): error CS0165: Use of unassigned local variable 'symmetricKeyMaterialLength' [D:\a\_work\1\s\src\libraries\System.Security.Cryptography\src\System.Security.Cryptography.csproj::TargetFramework=net9.0-windows]

```
https://dev.azure.com/dnceng-public/public/_build/results?buildId=747203&view=logs&j=d2b92d58-337d-5541-2ecb-d35f35f29352&t=3bcb3c60-9dac-5f6d-6a96-02987273246b

Comment (kunalspathak) [reply src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/Pbkdf2Implementation.Windows.cs:64]: Opened https://github.com/dotnet/runtime/issues/105126

**Comment by jkotas** [comment]
Reverted in #105128. @stephentoub Could you please resubmit?

**Comment by danmoseley** [comment]
Apologies, that was my fault (for not reverting).

**Pull request merged**